rust/datafusion/src/execution/context.rs

```diff
@@ -289,6 +290,37 @@ impl ExecutionContext {
                 let runtime_expr = self.create_physical_expr(expr, &input_schema)?;
                 Ok(Arc::new(SelectionExec::try_new(runtime_expr, input)?))
             }
+            LogicalPlan::Limit { input, expr, .. } => {
+                let input = self.create_physical_plan(input, batch_size)?;
+                let input_schema = input.as_ref().schema().clone();
+
+                match expr {
+                    &Expr::Literal(ref scalar_value) => {
+                        let limit: usize = match scalar_value {
+                            ScalarValue::Int8(x) => Ok(*x as usize),
```

Comment:
I think you need check that the signed integers are positive, right?

Reply:
Nice catch! ... this was copied over from the existing (soon to be legacy) query execution logic ... now fixed. Also used a more appropriate variable name than `x` and fixed the error messages.